Vector Cluster Changes

simon-leech opened the pull request:
Fixes #978 

If a layer is format cluster, and has no resolution or resolution is not number, it currently fails. 
This PR addresses the issue, checking for this and setting a default value and a console warning if so.

RobAndrewHurst commented:
FYI this was removed in this PR.

<img width="1500" alt="image" src="https://github.com/GEOLYTIX/xyz/assets/43169158/d480f94a-aa3c-4f50-b7d2-b87c7679d4db">

https://github.com/GEOLYTIX/xyz/pull/950/files

dbauszus-glx commented:
A vector layer may just have a cluster.distance instead of cluster.resolution.

dbauszus-glx reviewed (changes requested):
Format:cluster is a legacy format.

Checks where added to the vector format decorator which adds the params for `format:cluster` legacy layers.

It is completely valid to create a cluster vector layer with the params already configured. This is the desired configuration.

This breaks with this PR.

```js
"format": "vector",
"cluster": {
  "resolution": 0.2,
  "viewportLimit": 8
},
"params": {
  "viewport": true,
  "z": true,
  "template": "cluster"
},
```

simon-leech commented:
@dbauszus-glx Thanks for this, made some changes. 

1. If the layer has no distance or resolution, console.error and return. 
2. If the layer has cluster.resolution, then set the layer.params.viewport and layer.params.z to be true (needed for cluster queries).
3. If the layer has cluster.resolution, set layer.params.resolution to layer.cluster.resolution. 
4. If the layer has cluster.resolution, set layer.params.template to either params.template or cluster by default. 

cluster.hex_grid has been removed as unneeded, you can just pass layer.params.template="cluster_hex". 

Let me know if this is all ok, and I'll make sure the config docs are up to date!
``` js
"format": "vector",
"cluster": {
  "resolution": 0.2
},
"params": {
  "template": "cluster" // "cluster_hex"
}
```

simon-leech commented:
Added default srid of 3857 to vector format layers with layer.cluster.resolution layers, otherwise its 4326.
Added checks on srid too, updated other checks.

simon-leech commented:
Thanks for the help @RobAndrewHurst !
This is now ready for re-review 
@dbauszus-glx @RobAndrewHurst @eo-uk

dbauszus-glx reviewed (approved):
Please review my last commit.

The default SRID for all vector layers is 3857 and should be assigned first.

`layer.cluster?` doesn't need to be conditional within the if `layer.cluster` block.

The layer.cluster.resolution block should be within the layer.cluster block.

There shouldn't be two layer.cluster.resolution blocks.

And then finally, all the clusterConfig can be put into a separate method to clean up the vector format method.

https://github.com/GEOLYTIX/xyz/pull/981/commits/91e5734105fce6700d01510cb894a31d2045b201

simon-leech commented:
Thank you @dbauszus-glx ! 
I just made one change which is to check if cluster.resolution is numeric before performing `parseFloat()` on it - and warn if not. (I know its a bit silly for it to be non-numeric but always useful!)

simon-leech commented:
I've tried to update the config docs to explain these changes as simply as possible.
https://github.com/GEOLYTIX/xyz/wiki/Workspace-Configuration#vector-wkt-geojson-cluster
